tests/ha/single_pod_restart/test_server_pod_restart_api.py

```diff
@@ -768,3 +768,177 @@ def test_copy_obj_after_server_pod_restart(self):
+        self.restore_pod = False
+        event.clear()
+        LOGGER.info("Step 6: Checking responses from background process")
+        thread.join()
+        responses = tuple()
+        while len(responses) < 3:
+            responses = output.get(timeout=HA_CFG["common_params"]["60sec_delay"])
+        if not responses:
+            assert_utils.assert_true(False, "Background process failed to do copy object")
+        put_etag = responses[0]
+        exp_fail_bkt_obj_dict = responses[1]
+        failed_bkts = responses[2]
+        LOGGER.debug("Responses received from background process:\nput_etag: "
+                     "%s\nexp_fail_bkt_obj_dict: %s\nfailed_bkts: %s", put_etag,
+                     exp_fail_bkt_obj_dict, failed_bkts)
+        if len(exp_fail_bkt_obj_dict) == 0 and len(failed_bkts) == 0:
+            LOGGER.info("Copy object operation for all the buckets completed successfully. ")
+        elif failed_bkts or exp_fail_bkt_obj_dict:
+            assert_utils.assert_true(False, "Failed to do copy object when cluster was in degraded"
+                                            f" state. Failed buckets: \n{failed_bkts}"
+                                            f"\n{exp_fail_bkt_obj_dict}")
+        LOGGER.info("Step 6: Successfully completed copy object operation is background")
+        LOGGER.info("Step 7.1: Download the objects copied in healthy cluster and verify checksum")
+        for key, val in bkt_obj_dict.items():
+            resp = s3_test_obj.get_object(bucket=key, key=val)
+            LOGGER.info("Get object response: %s", resp)
+            get_etag = resp[1]["ETag"]
+            assert_utils.assert_equal(put_etag, get_etag, "Failed in Etag verification of "
+                                                          f"object {val} of bucket {key}. Put and "
+                                                          f"Get Etag mismatch")
+        LOGGER.info("Step 7.1: Download the objects copied in step 4 and verify checksum")
+        for key, val in bkt_obj_dict1.items():
+            resp = s3_test_obj.get_object(bucket=key, key=val)
+            LOGGER.info("Get object response: %s", resp)
+            get_etag = resp[1]["ETag"]
+            assert_utils.assert_equal(put_etag, get_etag, "Failed in Etag verification of "
+                                                          f"object {val} of bucket {key}. Put and "
+                                                          f"Get Etag mismatch")
+        LOGGER.info("Step 7: Successfully downloaded the object and verified the checksum")
+        t_t = int(perf_counter_ns())
+        LOGGER.info("Create and list buckets")
+        bkt_obj_dict.clear()
+        for cnt in range(bkt_cnt):
+            bkt_obj_dict[f"ha-bkt{cnt}-{t_t}"] = f"ha-obj{cnt}-{t_t}"
+        bucket_name = f"ha-mp-bkt-{t_t}-2"
+        LOGGER.info("Step 8: Perform copy object from %s bucket to other buckets verify copy "
```

Comment:
Same here. We are creating a new bucket. Either update the statement or set bkt_op to False, which will copy data from already created bucket to buckets in bkt_obj_dict